### DISCUSSION TITLE
Error on note recall (extension or browser incompatibility?)

### DISCUSSION THREAD

**Discussion by ghost** [coddingtonbear/obsidian-web#112] [Exotic Browser Support]
I've been getting errors on note recall. When I first installed the extension and Rest API plugin everything worked fine but after sometime for no apparent reason I can't append to my pre-existing notes, or check if I've already saved it.
When a URL (new or existing in obsidian) is opened it can not check if it already exists in notes but doesn't show `ERR` message either, when I reload the page it shows `ERR`. I can still add new notes, though, no problem with that.
I've tried to re-install extension, Rest API plugin, change settings, changing API key etc. none worked.

Extension version: 2.2.3
Local Rest API version: 1.5.2
Obsidian version: 1.1.16

**Comment by coddingtonbear** [comment]
What I'd recommend doing as a first troubleshooting step is going to your obsidian-web settings and turning *off* the note recall features, then turning them on one-by-one.  It could be that maybe the chrome permissions needed were revoked for some reason.

**Comment by ghost** [comment]
Turned off both search for previous note on tap and background and restarted chrome, still have same problem (shows ERR after reload of page!). It would be nice if we could see logs to see what's wrong. Not sure if it matters, but here's my exported config:

```json
{
  "sync": {
    "presets": [
      {
        "contentTemplate": "---\npage-title: {{json page.title}}\nurl: {{page.url}}\ndate: {{date \"yyyy-MM-dd\"}}\naliases: \ntags: \n- articles\n- todo\nauthor: \npublished: \nread: false\n---\n\n[Open]({{page.url}})\n\n## Notes & Highlights\n{{#if page.selectedText}}\n\n{{quote page.selectedText}}\n{{/if}}",
        "headers": {},
        "method": "put",
        "name": "Create new note",
        "urlTemplate": "/vault/Articles/{{filename page.title}}.md"
      },
      {
        "contentTemplate": "## {{page.title}}\nURL: {{page.url}}\n{{#if page.selectedText}}\n\n{{quote page.selectedText}}\n{{/if}}",
        "headers": {},
        "method": "post",
        "name": "Append to current daily note",
        "urlTemplate": "/periodic/daily/"
      },
      {
        "contentTemplate": "---\npage-title: {{json page.title}}\nurl: {{page.url}}\ndate: \"{{date}}\"\n---\n{{#if page.selectedText}}\n\n{{quote page.selectedText}}\n\n---\n\n{{/if}}{{page.content}}",
        "headers": {},
        "method": "put",
        "name": "Capture page snapshot",
        "urlTemplate": "/vault/{{filename page.title}}.md"
      },
      {
        "contentTemplate": "## {{date}}\n{{#if page.selectedText}}\n\n{{quote page.selectedText}}\n{{/if}}",
        "headers": {},
        "method": "post",
        "name": "Append to existing note",
        "urlTemplate": ""
      }
    ],
    "searchBackgroundEnabled": false,
    "searchEnabled": false,
    "searchMatchDirectTemplate": "Append to existing note",
    "searchMatchMentionTemplate": "Append to existing note",
    "version": "0.1"
  },
  "local": {
    "apiKey": "**********",
    "insecureMode": true,
    "version": "0.1"
  }
}
```

**Comment by coddingtonbear** [comment]
You might be able to see errors if you:

1. Right click on the extension icon
2. Click "Manage Extension"
3. Enabling "Developer mode" in the upper-right corner
4. Clicking on the link labeled "service worker" under "Inspect Views"

That's the inspector wrapping around the note recall feature.  While it's open, you might be able to try using note recall somewhere to see if any interesting errors appear.

Very often, that "ERR" display just means that it can't connect to Obsidian at all, though; so I'm surprised you'd see it in another situation.

**Comment by ghost** [comment]
enabled service worker inspect view with same configurations(search for previous note on tap and background off) and after clicking extension icon and reloading page received this logs:

```
background.js:1 TypeError: t.direct is not iterable
    at background.js:1:2955
    at Generator.next (<anonymous>)
    at i (background.js:1:1872)
(anonymous) @ background.js:1
i @ background.js:1
Promise.then (async)
s @ background.js:1
i @ background.js:1
Promise.then (async)
s @ background.js:1
(anonymous) @ background.js:1
(anonymous) @ background.js:1
(anonymous) @ background.js:1
```

**Comment by LightTemplar** [comment]
I have same problem. In the dev console for webpage there are multiple messages `checked runtime.lastError: The message port closed before a response was received.`
Dev Console for Service Worker shows this error:
```
background.js:1 TypeError: Failed to fetch
    at background.js:1:4823
    at Generator.next (<anonymous>)
    at background.js:1:3925
    at new Promise (<anonymous>)
    at background.js:1:3670
    at u (background.js:1:4582)
    at background.js:1:4402
    at Generator.next (<anonymous>)
    at background.js:1:3925
    at new Promise (<anonymous>)
(anonymous) @ background.js:1
c @ background.js:1
Promise.then (async)
s @ background.js:1
i @ background.js:1
Promise.then (async)
s @ background.js:1
(anonymous) @ background.js:1
(anonymous) @ background.js:1
(anonymous) @ background.js:1
```

I use  Brave (Version 1.51.110 Chromium: 113.0.5672.77 (Official Build) (64-bit) 
Ubuntu 22.04

**Comment by Dgaduin** [comment]
I'm having similar issues 
1. I get the this error from the `/search` endpoint
```
{
  "errorCode":40070
  "message":"The query you provided could not be processed.\ntag.replace is not a function"
}
```
2. This error on the console for inspecting the pop-up
```
Uncaught (in promise) TypeError: Q is not iterable
    at popup.js:1:9448
    at Generator.next (<anonymous>)
    at popup.js:1:6078
    at new Promise (<anonymous>)
    at i (popup.js:1:5823)
    at popup.js:1:9388
    at popup.js:1:9607
    at Dl (vendor.js:102:468081)
    at t.unstable_runWithPriority (vendor.js:102:499643)
    at Wa (vendor.js:102:408765)
(anonymous) @ popup.js:1
```
3.  And this one when loading the page 
```
 TypeError: t.direct is not iterable
    at background.js:1:2955
    at Generator.next (<anonymous>)
    at i (background.js:1:1872)
(anonymous) @ background.js:1
```

**Reply by esodesod** [reply]
I also had the errorCode 40070 and "The query you provided could not be processed.\ntag.replace is not a function", back in 2022. After verifying it was related to my specific notes (worked on a "clean vault"), I narrowed it down to an "invalid tag" in one of my notes causing the ERR in Obsidian Web/Local REST API. I can actually still reproduce:

If I put this exact line in YAML Front Matter/Obsidian Properties in any note, I get the ERR/40070 immediately.

```
tags: [33570]
```

The problem is the tag `33570`, which you actually won't be able to type into the new "Obsidian Properties" view, you have to edit using "source mode", or using another editor.

Back then, I simply removed the tag.

**Comment by ruben-sg** [comment]
Any news on this issue?

**Comment by coddingtonbear** [comment]
I'm afraid that I haven't seen this on any of my five environments that I use this in.  This could easily be caused by another incompatible browser extension, incompatibilities in the particular browser (the only one I've seen mentioned above is Brave, and I only test on Chrome and can't afford the time to spend testing on anything additional), or some other environmental factor I haven't even considered.

For whoever's running into this problem: what I'd recommend is creating a new browser profile in the latest version of Chrome *without* any other extensions.  I'd be surprised if it didn't work out-of-the-box when you're starting from such a clean slate, but if it did I'd be curious to hear more.  Assuming it did work, I'd recommend enabling extensions one-by-one until you identified which extension was causing the problem you're encountering.

If it helps at all, the two browser versions I'm right in front of right now that use this without trouble include:

- Chrome: 114.0.5735.198 (Official Build) (64-bit)
- Chrome: 115.0.5790.102 (Official Build) (64-bit)

and although the version of Chrome probably doesn't matter much, I really need to stress that you should use Chrome itself (not Brave) to get to an actual "clean slate".

Since I haven't been able to reproduce this and am fairly sure that this is an incompatibility in an unsupported configuration, I'm going to convert this to a discussion until some bug in this extension is identified.  That's not the end of this conversation, though -- if somebody does figure out what the incompatibility is, I'd love to review a PR that adapts this extension to make it more compatible with unsupported environments like Brave.

**Comment by ruben-sg** [comment]
Thanks for your reply. I tried what you recommended. But the same happens. And seeing that there are more users experiencing this error (on reddit, for example), I guess there is some problem. I have no idea what could cause it. But, it semed somewhat weird that none of the fields are filled in when trying to append (see screenshot):
![image](https://github.com/coddingtonbear/obsidian-web/assets/24955840/578dd66f-cb4a-43b8-8ad9-8acb55caadc3)

**Reply by coddingtonbear** [reply]
I'm not aware of any reddit threads or subreddits in which this has been discussed, and I'm curious to hear about ... where...  even if I'm not able to commit time to keeping up with them.

**Comment by ruben-sg** [comment]
Fully understand that you are not able to work on that 24/7 :) If it is of help, two links:
- https://www.reddit.com/r/ObsidianMD/comments/y7qnpu/has_anyone_gotten_obsidian_web_to_work/
- https://www.reddit.com/r/ObsidianMD/comments/1063e6r/what_do_you_use_to_take_notes_while_browsing/

**Comment by ruben-sg** [comment]
I have found the solution (and I'm sorry for having bothered you, it's that obvious). When appending, there has to be an entry in the field API Url when POSTing that is pointing to the exact note.

**Reply by coddingtonbear** [reply]
I don't think, actually, that you should have to do that -- when you press the up arrow, it _should_ populate the URL for the note automatically.

**Reply by coddingtonbear** [reply]
But, just to be clear, I don't know that's quite the same kind of problem that the original post above is noting; so you may want to open up a separate discussion for that.

**Reply by connradolisboa** [reply]
Can you explain how did you manage to solve this? The extension never recalls any note inside my vault and if I send a highlight it onlys rewrites the page

**Reply by ruben-sg** [reply]
to be perfectly honest: I cannot remember. I stopped using the plugin.
